game/hud/src/widgets/Compass/index.tsx

```diff
@@ -64,11 +79,20 @@ class Compass extends React.Component<CompassProps, CompassState> {
   }
 
   render() {
+    if (!this.state.visible) return null;
```

Comment:
this is not doing visibility of a part of the widget, this is doing visibility for the entire widget which should be handled externally so we're not even rendering the widget.